flink-runtime/src/test/java/org/apache/flink/runtime/rest/compatibility/CompatibilityRoutines.java

```diff
@@ -230,6 +230,7 @@ private static void assertCompatible(final JsonNode old, final JsonNode cur) {
                 final JsonNode curProperties = curProperty.get("properties");
 
                 if (oldProperties != null) {
+                    assertThat(curProperties).as("New object has no properties.").isNotNull();
```

Comment:
This was necessary because the compatibility checks run in both directions, and we never accounted for the possibility of an endpoint returning `object` at one point and then switching to `any`.